### PR TITLE
Introduce year grid view (with FC multiMonthYear plugin)

### LINKED ISSUE #159: year view
It would be really great to have a "year view" in addition to day/week/month. I have many use-cases, e.g. "vacation days/weeks", or "time I am in Berlin" or "time the project room is in use".... often it just really helps to be able to see which weeks have something set and which don't. It is quite cumbersome to click through the months....
Thank you very much!

PS: this was already suggested some time ago, by someone else: https://forum.owncloud.org/viewtopic.php?t=2127

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38782135-year-view?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by tcitworld**
Closes #159

image

Note: Shortcut for ListView is changed from 4 to 5 for consistency.

It could use a little style enhancement.

FullCalendar provides two "year" views: [this one (multiMonthYear)](https://fullcalendar.io/docs/multimonth-grid-demo) and [another one (dayGridYear)](https://fullcalendar.io/docs/daygridyear-view-demo). This one seems more suitable.

### CODE REVIEW

File: src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderDatePicker.vue [hunk 1]
```diff
@@ -64,7 +65,7 @@ import {
 	modifyDate,
 } from '../../../utils/date.js'
 import { mapState } from 'vuex'
-import formatDateRage from '../../../filters/dateRangeFormat.js'
```
Comment (ChristophWurst) [inline src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderDatePicker.vue:67]: ngl working with dates sometimes ends in rage

**Review by ChristophWurst — APPROVED** [review]
Code looks clean

**Review by miaulalala — APPROVED** [review]

**Comment by miaulalala** [comment]
needs a rebase, but other than that

image

**Comment by tcitworld** [comment]
rebased

**Comment by miaulalala** [comment]
Sorry, can you rebase once more? Static Analysis hasn't run because of timing.

**Comment by bernd-wechner** [comment]
Awesome! Cannot wait to see this. Whenever possible let us know which version it appears in (nominally 4.5.0 of the Calendar app). Job well done! Been waiting on this for years.

**Pull request merged**